File: 프로그래머스/1/42840. 모의고사/모의고사.java
class Solution {
    public int[] solution(int[] answers) {
        int[] answer = {};
        int[] person1 = {1, 2, 3, 4, 5};
        int[] person2 = {2, 1, 2, 3, 2, 4, 2, 5};
        int[] person3 = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5};
        
        int[] correct = new int[3];
        
        for(int i=0; i<answers.length; i++){
            if(answers[i] == person1[i%5]) correct[0]++;
            if(answers[i] == person2[i%8]) correct[1]++;
            if(answers[i] == person3[i%10]) correct[2]++;
        }
        
        int maxScore = 0;
        for(int i=0 ; i<3; i++) maxScore = Math.max(maxScore, correct[i]);
        int count = 0;
        for(int i=0 ; i<3; i++) if(maxScore == correct[i]) count++;
        int[] best = new int[count];
        int tmp = 0;
        for(int i=0; i<3; i++) {
            if(maxScore == correct[i]){
                best[tmp] = i+1;
                tmp++;
            }
        };
        
        return best;
    }
}

File: 프로그래머스/1/42840. 모의고사/모의고사.js
function solution(answers) {
    var answer = [];
    const first = [1,2,3,4,5];
    const second = [2,1,2,3,2,4,2,5];
    const third = [3,3,1,1,2,2,4,4,5,5];
    
    let firstC = 0;
    let secondC = 0;
    let thirdC = 0;
    
    for(let i = 0; i < answers.length; i++){
        if(first[i%5] == answers[i]) firstC++;
        if(second[i%8]==answers[i]) secondC++;
        if(third[i%10]==answers[i]) thirdC++;
    }
    
    let highest = Math.max(firstC, secondC, thirdC);
    
    
    console.log(firstC);
    console.log(secondC);
    console.log(thirdC);
    
    if(firstC == highest) answer.push(1);
    if(secondC == highest) answer.push(2);
    if(thirdC == highest) answer.push(3);
    
    return answer;
}